src/Message/EnvelopeTrait.php

```diff
@@ -21,9 +21,12 @@ public function withMessage(MessageInterface $message): self
         return $instance;
     }
 
-    public function getHandlerName(): string
+    /**
+     * @return class-string<MessageHandlerInterface>
+     */
+    public function getHandler(): string
     {
-        return $this->message->getHandlerName();
+        return $this->message->getHandler();
```

Comment:
There may be a problem here, when handler support only original message and don't support envelope.